libr/reg/reg.c

```diff
@@ -119,7 +122,7 @@ R_API void r_reg_free(RReg *reg) {
 		return;
 
 	for (i = 0; i < R_REG_TYPE_LAST; i++) {
-		r_list_purge (reg->regset[i].pool);
+		r_list_free (reg->regset[i].pool);
```

Comment:
uhm i think there was a reason to use purge instead of free here..cant remember now